File: python/e_1-50/e_1-5/e_2_even_fib.py
def fib_upto(n: int) -> list[int]:
    """
    Generate a list of Fibonacci numbers up to a given limit.

    Args:
        n (int): The limit up to which Fibonacci numbers should be generated.

    Returns:
        list[int]: A list of Fibonacci numbers up to the given limit.
    """
    list = [0] * n
    list[0] = 1
    list[1] = 1
    for i in range(2, n):
        list[i] = list[i-1] + list[i-2]
    return list

print(sum(n for n in fib_upto(100) if n % 2 == 0 and n < 4000000))

File: typescript/e_1-50/e_1-5/e_2_even_fib.ts
/**
 * Generates Fibonacci sequence up to a given number.
 * @param n - The number up to which the Fibonacci sequence should be generated.
 * @returns An array containing the Fibonacci sequence up to the given number.
 */
function fib_upto(n: number): number[]{
    let fin: number[] = []
    fin.push(1)
    fin.push(1)
    for (let i = 2; i < n; i++)
        fin.push(fin[i-1] + fin[i-2])
    return fin
}

let fibs: number[] = fib_upto(100)
let s: number = 0
for (let i = 0; i < fibs.length; i ++) {
    if (fibs[i] % 2 === 0 && fibs[i] < 4000000)
        s += fibs[i]
}
console.log(s)
